src/crewai/agent.py

```diff
@@ -174,7 +245,38 @@ def execute_task(
 
         Returns:
             Output of the agent
+
+        Raises:
+            TimeoutError: If the task execution exceeds max_execution_time (if set)
+            Exception: For other execution errors
         """
+        if self.max_execution_time is None:
+            return self._execute_task_without_timeout(task, context, tools)
+
+        original_llm_timeout = getattr(self.llm, 'timeout', None)
+        try:
+            if hasattr(self.llm, 'timeout'):
+                self.llm.timeout = self.max_execution_time
+            
+            return self._execute_with_timeout(task, context, tools, self.max_execution_time)
+        except timeout_decorator.TimeoutError:
+            error_msg = (
+                f"Task '{task.description}' execution timed out after {self.max_execution_time} seconds. "
+                f"Consider increasing max_execution_time or optimizing the task."
+            )
+            self._logger.log("error", error_msg)
```

Comment:
After updating your branch  with main take a [look here ](https://github.com/crewAIInc/crewAI/blob/fc9da22c38674a1daae7b3776dafb3e8bc9d4032/src/crewai/agent.py#L258C17-L265C18). Would be great ensure the timeout event will be sent